### PR TITLE
Refactor Stiefel

### PR DESCRIPTION
**Pull request by luisfpereira**
This PR refactors Stiefel by removing the `log`  computation to an external object (a solver). This way, numerical parameters are easy to control and we can implement new algorithms to compute the log. Additionally, `lifting` is fixed (it was not working) and a tolerance is added to a test that fails recurrently.

It should be said that the `log` has problems, as during the iterative process it starts getting complex numbers as result. This comes from `logm`. We should address this is a new PR.

### CODE REVIEW

**Comment by ninamiolane** [comment]
Some tests, related to Stiefel, are failing: is this expected?

**Comment by luisfpereira** [comment]
It was related with tolerances. I've adjusted them.

**Review by ninamiolane — APPROVED** [review]
LGTM! Thank you very much!

**Review by ninamiolane — COMMENTED** [review]
One comment on why we have space as input parameter of the log.

File: geomstats/geometry/stiefel.py [hunk 1]
```diff
@@ -343,16 +498,18 @@ def _procrustes_preprocessing(p, matrix_v, matrix_m, matrix_n):
...
             mask = gs.cast(det < 0, matrix_v.dtype)
             sign = mask[..., None] * reflection_vec + (1.0 - mask)[..., None] * ones
             matrix_d = gs.einsum(
                 "...ij,...i->...ij", Matrices.transpose(matrix_d), sign
             )
+
         return matrix_v_final
 
-    def log(self, point, base_point, max_iter=30, tol=gs.atol, **kwargs):
+    def log(self, space, point, base_point):
```
[the remark this answers is not in the document]
Comment (luisfpereira) [reply geomstats/geometry/stiefel.py:512]: This is the `log` implemented by a solver, which by design does not have access to the space at instantiation.

**Pull request merged**